src/util/address.rs

```diff
@@ -499,6 +538,7 @@ impl Address {
                         32 => Some(AddressType::P2wsh),
                         _ => None,
                     },
+                    WitnessVersion::V1 if prog.len() == 32 => Some(AddressType::P2tr),
```

Comment:
I guess better for a different PR but I'd like to see this `fn` return `Result` with more information in `Err` - in this case invalid length.

Reply:
You are right, and actually fixing that led me to discover another hidden error: some BIP350 test vecs are correct as Bech32m, but are incorrect as address representation. So I have added new error types, required checks etc in the last separate commit to this PR